GoogleForCreators/web-stories-wp
# Media: Improve plugin compatibility of taxonomy query

## Context

We are using folders plugin to structure media files in folders, web stories plugin creates invalid SQL query which does not provide any results

## Summary

When eb Stories plugin is active and there is no any `$tax_query` in `WP_Query` object it adds `0 = 1` in SQL query which is always wrong so not returning any results in `WP_Query`, Please check this video for more details: https://tinyurl.com/2anx49r9

## Relevant Technical Choices

We have fixed issue in this file:
`/wp-content/plugins/web-stories/includes/Media/Media_Source_Taxonomy.php` added this code after line number: 382

```php
if(empty($tax_query)) {
    return [
        [
            'taxonomy' => $this->taxonomy_slug,
            'field'    => 'slug',
            'terms'    => [
                self::TERM_POSTER_GENERATION,
                self::TERM_SOURCE_VIDEO,
                self::TERM_SOURCE_IMAGE,
                self::TERM_PAGE_TEMPLATE,
            ],
            'operator' => 'NOT IN',
        ],
    ];
}
```

## To-do

Any filter functionality on WP media page

## User-facing changes

Before: https://tinyurl.com/25j68dyu
After: https://tinyurl.com/23c7k7kd

## Testing Instructions
Install Folders plugin from wordpress.org (https://wordpress.org/plugins/folders/)
After Installing folders on your wordpress website, go to media and create folders and assign some media items to that folders (check this video for how to create folders and assign media items to it) - https://tinyurl.com/2anx49r9 then click on any Folders it is not showing any results 

This PR can be tested by following these steps:

1. Create folders and assign some media items into it and then click on folders or follow the steps as shown in video (https://tinyurl.com/2anx49r9)

## Reviews

### Does this PR have a security-related impact?

No

### Does this PR change what data or activity we track or use?

No

### Does this PR have a legal-related impact?

No

## Checklist




---
